cfme/tests/containers/test_smart_management.py

```diff
@@ -0,0 +1,93 @@
+import random
+
+import pytest
+
+from cfme.common.provider import BaseProvider
+from cfme.containers.container import Container
+from cfme.containers.image_registry import ImageRegistry
+from cfme.containers.node import Node
+from cfme.containers.pod import Pod
+from cfme.containers.project import Project
+from cfme.containers.provider import ContainersProvider
+from cfme.containers.provider import ContainersTestItem
+from cfme.containers.template import Template
+from cfme.utils.blockers import BZ
+from cfme.utils.wait import wait_for
+
+pytestmark = [
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.tier(1),
+    pytest.mark.provider([ContainersProvider], scope='function')
+]
+
+TEST_ITEMS = [
+    pytest.mark.polarion('CMP-9948')(ContainersTestItem(
+        Container, 'CMP-9948', collection_obj="containers", keyword_name="name")),
+    pytest.mark.polarion('CMP-10320')(ContainersTestItem(
+        Template, 'CMP-10320', collection_obj="container_templates", keyword_name="name")),
+    pytest.mark.polarion('CMP-9992')(ContainersTestItem(
+        ImageRegistry, 'CMP-9992', collection_obj="container_image_registries",
+        keyword_name="host")),
+    pytest.mark.polarion('CMP-9964')(ContainersTestItem(
+        Node, 'CMP-9964', collection_obj="container_nodes", keyword_name="name")),
+    pytest.mark.polarion('CMP-9932')(ContainersTestItem(
+        Pod, 'CMP-9932', collection_obj="container_pods", keyword_name="name")),
+    pytest.mark.polarion('CMP-9870')(ContainersTestItem(
+        Project, 'CMP-9870', collection_obj="container_projects", keyword_name="name")),
+    pytest.mark.polarion('CMP-9854')(ContainersTestItem(
+        ContainersProvider, 'CMP-9854', collection_obj="containers_providers",
+        keyword_name="name"))
+]
+
+
+def get_entity(collection):
+
+    # Map all object by name
+    mapping = {item.name: item for item in collection.all()}
+
+    # Filter only the unique items
+    unique_items = [key for key in set(mapping.keys()) if mapping.keys().count(key) == 1]
+
+    # If there is at least one  unique item
+    if unique_items:
+        # Return a random item
+        selected_item_name = random.choice(unique_items)
+        return mapping[selected_item_name]
+    else:
+        pytest.skip("No unique item was found")
+
+
+def get_clean_entity(collection=None, entity=None):
+    if collection is None and entity is None:
+        pytest.fail('Needed to pass a collection or entity within test')
+    else:
+        entity = entity or get_entity(collection)
+
+    tags = entity.get_tags()
+    entity.remove_tags(tags)
+    return entity
+
+
+@pytest.mark.parametrize('test_item', TEST_ITEMS,
+                         ids=[ContainersTestItem.get_pretty_id(ti) for ti in TEST_ITEMS])
+@pytest.mark.meta(blockers=[BZ(1609541)])
+def test_smart_management_add_tag(provider, appliance, test_item):
+
+    # validate no tag set to project
```

Comment:
Optional:
Better to call `entity` or `object`. Project scope is much as per me.